### ISSUE TITLE
Possible change in location for the packages in R-Instat?

### ISSUE THREAD

**Issue by rdstern** [africanmathsinitiative/R-Instat#7064]
One powerful feature in R-Instat is in the Tools Menu where you can import a new package into R-Instat.  The dialogue is useful and I often also use it to check whether a particular package is already installed.  If so, is it up-to-date.
This works fine on the development (merged) version of R-Instat.  It is also a very nice feature (even nicer!) in RStudio - as we might expect.
However in the "proper" version of R-Instat it isn't quite so nice.  You have to be an administrator to install it.  That's not difficult to arrange on my machine, but it isn't so easy for everyone, and it isn't needed in RStudio.  Could we store the packages in a different location - on installation - so packages could be added that bit more easily?
 I think improved installation is to be considered soon, and perhaps this element could be included.  I find the current installation is pretty easy, but the pain is that we have no upgrading facility.  You just have to download and install the whole of the next version and then delete the old one.  Almost all other software does upgrading more smoothly.  (This is important for us, because we claim to be appropriate for countries with limited internet access.  We are mostly, but expecting people to upload 700 mbytes each time is an exception.)

**Comment by rdstern** [comment]
@chris, this isn't urgent, but I raise it again, so it isn't forgotten.  I assume this will be quite an easy change?  Could it be done by @Patowhiz?  I was assuming it had to involve you?

**Comment by rdstern** [comment]
@lloyddewit and @ChrisMarsh82 just asking again on the location of the R packages in R_Instat, as packages are being discussed now.  I note that with RStudio it is easy to add a package.  In R-Instat (released version), they can only be added if you have administrator rights.  It would be good if that didn't have to be the case.

**Comment by Patowhiz** [comment]
In windows platform, there are specific directories(in AppData directory) designated for apps to store their data. These directories don't require administrator rights. Softwares like web browsers use them.

I'd suggest that to be the default folder path. It's standard for an app targeting windows to use that.

**Comment by ChrisMarsh82** [comment]
@Patowhiz is right. We are currently installing R and all its packages in Program Files. To do this requires administrator access. We could move all the R packages to AppData and maybe even R.
To be considered:
R-studio links to whatever version of R you have installed. R-Instat deliberately doesn't do this so that we can be in full control of the version of R and its packages. We need to make sure we do not conflict with other versions of R installed and other packages installed if we move out of program files.
Can users break existing functionality by installing a package? Could one package overwrite another? If so keeping it behind admin rights might not be a bad idea.
Currently, we don't upgrade R-Instat, we install a new instance. We do this but share AppData (i.e. Options). This could make AppData a bit messy as we will need individual folders for each version plus a shared folder for our existing AppData. 
We would need to make sure uninstall deletes the relevant AppData

@dannyparsons @volloholic - Is there a deliberate design choice I may be missing why R and its packages are installed into program files?

**Comment by rdstern** [comment]
@ChrisMarsh82 many thanks.  I'll follow up here with Danny and maybe David too.

**Comment by rdstern** [comment]
@ChrisMarsh82 and @lloyddewit I am getting ready to close this issue, but want to put on record exactly why we propose, at least for now, to keep the location of the R-Instat files where they are, i.e. so you will need adminstrator prividges to add a new package into R-Instat.

So currently the packages are installed in program files.  They are also within a directory associated with a particular version of R-Instat.  So, you may have 2 versions of R-Instat on your machine.  If so, they will each have a separate library.  You can run the 2 versions separately if you wish.  I think that's true, and I claim I have been doing that.

Also, you may have RStudio on your machine.  If so, the packages in R-Instat are (usually) totally separate from those in RStudio.  

We therefore have a simple system.  It is a bit of a waste of space and a bit annoying to have to be an administrator to install a package.  But that is a very advanced feature in R-Instat, so that is not a great worry to me.

I can associate a file type with a particular software, and I think @N-thony has arranged that it is therefore possible to associate (a version of) R-Instat with an R file or an RDS file.  I am not clear what this means for running multiple instances of R-Instat with the same version of the software.  I'll check with @N-thony because I need to be clear on this for writing the help files.

OK, so far?

This also relates (I think) to another topic discussed yesterday, (21 September 2023) namely to only offer full installations of R-Instat and not updates.  So I take the opportunity to make sure I understand the implications of both these decisions.

Chris informs me that this is not as unusual as I thought.  So, at least for now, we will not offer upgrades.  If we routinely go to upgrades, then that becomes very confusing if we would like to alos have 2 (or more) versions of the software.  

@ChrisMarsh82 we did not discuss the issue of serious bugs introduced in an upgrade.  In that situation we would prfer users to not use that version - or perhaps to upgrade it.  Would it be an easy step to offer bug-fixing updates in addition to our usual new versions?  And that might relate to the numbering.  

By the way, the number is currently not provided.  I am thinking that we are aiming towards 0.8.0, 0.8.1, etc.  Though not yet - the next one is 0.7.17 I think. Should we consider allowing for 0.7.17a and b, etc, but only if critical and needed.  And if you install b, then it includes a, or perhaps not.  I hope it will be rare that you even need a!  Then 0.7.18 is a full install as usual.

**Comment by lloyddewit** [comment]
@rdstern 
> OK, so far?

Yes, I checked on my machine and I agree with what you state above.
Note: On my machine, if I run a development version of R-Instat (i.e. from Visual Studio), then it will use the R version installed in `C:\Program Files\R`. RStudio also uses this version so if I install a package in RStudio, then my R-Instat development version has access to the same package.
